src/App.js

```diff
@@ -1,23 +1,45 @@
 import React, { Component } from 'react';
 import {Switch, Route, HashRouter} from 'react-router-dom';
+import firebase from './firebase';
+
+// Pages
 import Trip from './containers/Trip/Trip';
 import Navbar from './components/Navbar/Navbar';
 import Home from './components/Home/Home';
 
+// Context
+import FirebaseAuthContext from './context/FirebaseAuth';
+
 class App extends Component {
+    state = {
+        user: null
+    }
+
+    componentDidMount () {
+        this.unsubscribe = firebase.auth().onAuthStateChanged(user => {
+            if(user){
+                this.setState({ user });
+            } else {
+                this.setState({ user: null });
+            }
+        })
+    }
+
     render () {
         return (
-            <div className="App">
-                <HashRouter>
-                    <Route path='/' component={Navbar} />
-                    <div>
-                        <Switch>
-                            <Route path='/trip' exact component={Trip}></Route>
-                            <Route path='/' exact component={Home} />
-                        </Switch>
+            <HashRouter>
+                <FirebaseAuthContext.Provider value={this.state.user}>
```

Comment:
Your `App` component is getting a couple of large responsibilities:

1. Routing
2. Authentication state

Since nothing else in this file uses `state.user`, it might be easier to move the `<FirebaseAuthContext.Provider` and state to a separate component that sets context. Have you tried that?

Reply:
I haven't tried that yet. I was wrapping the FirebaseAuthContext.Provider around the children, so the other components, will import the FirebaseAuthContext.Consumer and if the value user exists, show either a private component or Redirect to Home. Something like this:
``` javascript
<AuthContext.Consumer>
                {
                    ({user}) => {
                        if(user){
                            return <h1>Hello Private Component</h1>
                        } else {
                            return <Redirect to='/' />
                        }
                    }
                }
            </AuthContext.Consumer>
```